release-note: settings, management and other updates (#28221)
zephyrproject-rtos/zephyr: doc/releases/release-notes-2.4.rst

* EEPROM

  * Added driver supporting the on-chip EEPROM found on NXP LPC11U6X MCUs.

>   * Fix at2x cs gpio flags extracion from DT.

Reviewer: ```suggestion
  * Fixed at2x cs gpio flags extraction from DT.
```

  * Allow user to disable auto-start of IEEE 802.15.4 network interface.
    By default the IEEE 802.15.4 network interface is automatically started.
  * Added support for setting TX power in rf2xx driver.

>   * nordic: add 802.15.4 multiprotocol support, see :option:`CONFIG_NRF_802154_MULTIPROTOCOL_SUPPORT`.

Reviewer 2: Looks a bit different than the other lines. Perhaps `* Added Nordic 802.15.4 multiprotocol ...`

>   * Added Kconfig :option:`CONFIG_IEEE802154_VENDOR_OUI_ENABLE` option f

Reviewer 2: What is the `f` at the end of the line?

* Fixed LWM2M Security and Server object instance matching.
* Fixed LWM2M updating lifetime on Register Update event.
* Fixed MQTT double CONNACK event notification on server reject.

> * PPP: Added initial support for PAP authentication.

Reviewer 2: Please add this to "Added" section above after the other PPP option. So something like this
```
* Added support to rejecting received and unsupported PPP options.
* Added initial support for PAP authentication.
```

> * PPP: Automatic select CONFIG_RING_BUFFER as it is required.

Reviewer 2: IMHO no need to mention this in the release note. I went through all the changes and only picked relevant stuff and not all changes. It seems I missed some important stuff like the PAP support.

> * PPP: Refactored option negotiation.

Reviewer 2: IMHO no need to mention this in release note.

> * PPP: Fixed a race condition when ppp_fsm_open() was called in CLOSED state.

Reviewer 2: This could be placed after other fixed PPP stuff above. So something like
```
* Fixed a race PPP when ppp_fsm_open() was called in CLOSED state.
```

> * PPP: Fsm: use packet work object instead of shared one, which allow to handle more than one packet at one time.

Reviewer 2: IMHO no need to mention this in release note.

* Disk

* Management

  * MCUmgr:

    * Moved mcumgr into its own directory.
    * UDP port switched to using kernel stack.
    * smp: added missing socket close in error path.

  * Added support for Open Supervised Device Protocol (OSDP), see :option:`CONFIG_OSDP`.

>   * updatehub: Move from lib to subsys:mgmt directory.

Reviewer 2: Should we have `subsys/mgmt` so `/` instead of `:`

    * Moved updatehub from lib to subsys:mgmt directory.

>     * Fix out-of-bounds access and add flash_img_init return value check.

Reviewer 2: s/Fix/Fixed/

>     * Fix getaddrinfo resource leak.

Reviewer 2: Same here

* Settings:

>   * If a setting read is attempted from a channel that doesn't support reading return an error rather than faulting.

Reviewer 2: There should be empty line after `* Settings:`

* Power management:

* Logging:

>   * fix immediate logging with multiple backends.

Reviewer: ```suggestion
  * Fixed immediate logging with multiple backends.
```

>   * Switch logging thread to use kernel stack.

Reviewer: ```suggestion
  * Switched logging thread to use kernel stack.
```

Tests and Samples
*****************

>   * nvs: Do full chip erase when flashing.
>   * nrf: onoff_level_lighting_vnd_app: fix build with mcumgr.
>   * drivers: flash_shell: new commands write_unaligned and write_pattern.
>   * bluetooth: hci_spi: Fix cmd_hdr and acl_hdr usage.
>   * remove useless zephyr nfc sample.
>   * drivers: Fix unitialized spi_cfg in spi_fujitsu_fram sample.
>   * Updated configuration for extended advertising in Bluetooth hci_uart and hci_rpmsg examples.

Reviewer: ```suggestion
  * nvs: Do full chip erase when flashing.
  * nrf: onoff_level_lighting_vnd_app: Fixed build with mcumgr.
  * drivers: flash_shell: new commands write_unaligned and write_pattern.
  * bluetooth: hci_spi: Fixed cmd_hdr and acl_hdr usage.
  * Removed zephyr nfc sample.
  * drivers: Fixed uninitialized spi_cfg in spi_fujitsu_fram sample.
  * Updated configuration for extended advertising in Bluetooth hci_uart and hci_rpmsg examples.
```
